Platform Installer

DidierRLopes opened the pull request.

DidierRLopes committed:
initial version

DidierRLopes committed:
update content

deeleeramone committed:
introduction --> index + add self-signed cert steps

deeleeramone committed:
Merge branch 'main' into feat/platform-installer

DidierRLopes committed:
Merge branch 'main' into feat/platform-installer

deeleeramone committed:
Merge branch 'main' into feat/platform-installer

deeleeramone committed:
Merge branch 'main' into feat/platform-installer

deeleeramone committed:
add odp category

deeleeramone committed:
add screenshots

deeleeramone committed:
update backends

deeleeramone committed:
Add screenshots

deeleeramone committed:
renames file extension as mdx

deeleeramone committed:
some fixes

deeleeramone committed:
index item

deeleeramone committed:
minor updates

deeleeramone committed:
fix sentence

deeleeramone committed:
Merge branch 'main' into feat/platform-installer

deeleeramone committed:
environments page

deeleeramone committed:
add screenshots...

deeleeramone committed:
fix bad console icon screenshots

deeleeramone committed:
merge branch main

deeleeramone committed:
api keys page

deeleeramone committed:
add screenshots

deeleeramone committed:
add uninstall

deeleeramone committed:
add screenshot

deeleeramone committed:
rename file back to mdx

deeleeramone committed:
Merge branch 'main' into feat/platform-installer

piiq commented:
@deeleeramone is this one outdated or ready to review? Or is this waiting for some announcement?

DidierRLopes commented:
I would merge this one, and have it get some mileage. Potentially some folks could give some feedback to it early.

@piiq

DidierRLopes committed:
Merge branch 'main' into feat/platform-installer

deeleeramone committed:
Merge branch 'main' of https://github.com/OpenBB-finance/openbb-docs into feat/platform-installer

DidierRLopes committed:
Merge branch 'main' into feat/platform-installer

piiq commented:
@deeleeramone can this be merged?

deeleeramone commented:
No. This cannot be merged yet.

deeleeramone committed:
Merge branch 'feat/platform-installer' of https://github.com/OpenBB-finance/openbb-docs into feat/platform-installer

deeleeramone committed:
Update a couple images

deeleeramone committed:
Merge branch 'feat/platform-installer' of https://github.com/OpenBB-finance/openbb-docs into feat/platform-installer

deeleeramone committed:
commit some stuff

deeleeramone committed:
Add screenshots

deeleeramone committed:
install screenshots

deeleeramone committed:
temp commit

deeleeramone committed:
screenshots

Updated examples and instructions related to API keys, including changes to displayed text and added images.

DidierRLopes committed:
Merge branch 'main' into feat/platform-installer

deeleeramone committed:
refactor for odp

deeleeramone committed:
Merge branch 'main' of https://github.com/OpenBB-finance/openbb-docs into feat/platform-installer

deeleeramone committed:
Merge branch 'feat/platform-installer' of https://github.com/OpenBB-finance/openbb-docs into feat/platform-installer

deeleeramone committed:
remove charting.md from developer for now

deeleeramone committed:
fix some links

piiq commented:
@deeleeramone how about now? =)

deeleeramone committed:
Merge branch 'main' into feat/platform-installer

piiq commented:
@deeleeramone I've ran the `scripts/generate_platform_markdown.py` in the environment where I have the latest `openbb` installed from the develop branch with all the extensions and the cli and I still get the following broken links:

```
[WARNING] Docusaurus found broken links!

Please check the pages of your site in the list below, and make sure you don't reference any path that does not exist.
Note: it's possible to ignore broken links with the 'onBrokenLinks' Docusaurus configuration, and let the build pass.

Exhaustive list of all broken links found:
- Broken link on source page path = /cli/routines:
   -> linking to routines/community-routines (resolved as: /cli/routines/community-routines)
- Broken link on source page path = /desktop/api-keys:
   -> linking to /python/settings/user_settings
- Broken link on source page path = /desktop/backends:
   -> linking to /odp/environments
- Broken link on source page path = /desktop/installation:
   -> linking to /odp/environments
- Broken link on source page path = /python:
   -> linking to /python/settings
- Broken link on source page path = /python/basic_usage:
   -> linking to /python/models
- Broken link on source page path = /python/basic_usage/response_model:
   -> linking to /python/settings/user_settings
- Broken link on source page path = /python/developer/architecture_overview:
   -> linking to extensions (resolved as: /python/developer/extensions)
- Broken link on source page path = /python/developer/extension_types/obbject:
   -> linking to /python/developer/extension_types/charting
- Broken link on source page path = /python/extensions/interface/openbb-mcp:
   -> linking to /workspace/developers/tools/openbb-api
- Broken link on source page path = /python/faqs/data_providers:
   -> linking to /platform/developer_guide/misc/contributing
   -> linking to /platform/reference
   -> linking to /platform/usage/extensions/overview
- Broken link on source page path = /python/installation:
   -> linking to /odp/installation
- Broken link on source page path = /python/quickstart/rest_api:
   -> linking to /python/settings/user_settings/credentials
- Broken link on source page path = /python/quickstart/workspace:
   -> linking to /python/extensions/openbb-api
- Broken link on source page path = /workspace/getting-started/platform-installer:
   -> linking to /platform/settings/user_settings/api_keys

[WARNING] Docusaurus found broken anchors!

Please check the pages of your site in the list below, and make sure you don't reference any anchor that does not exist.
Note: it's possible to ignore broken anchors with the 'onBrokenAnchors' Docusaurus configuration, and let the build pass.

Exhaustive list of all broken anchors found:
- Broken anchor on source page path = /python:
   -> linking to /desktop#installation
- Broken anchor on source page path = /python/settings/user_settings/preferences:
   -> linking to api_keys#local-environment (resolved as: /python/settings/user_settings/api_keys#local-environment)
```

Am I missing anything or are these links actually broken?

piiq commented:
Just tested one and it's a real 404
<img width="1270" height="443" alt="image" src="https://github.com/user-attachments/assets/298a7954-f1fa-4704-b26f-a8f50001538b" />

piiq committed:
Bump axios to a patched version

deeleeramone committed:
fix some broken links

deeleeramone commented:
> - Broken link on source page path = /desktop/api-keys:
>    -> linking to /python/settings/user_settings

How about now?

deeleeramone committed:
one more

deeleeramone committed:
Replace uninstall image with updated version

Updated uninstall image to a new source with different dimensions.

deeleeramone committed:
Replace certificate creation image with new one

Updated image for certificate generation section.

piiq committed:
Update "OpenBB Platform" -> "ODP" naming in documentation and keywords

piiq committed:
Lint markdown and fix typos

piiq committed:
Change Jinja style comments to markdown style comments

piiq committed:
Prevent `generate_plarform_markdown.py` from creating both file and folder references

piiq committed:
Lint `generate_platform_markdown.py`

piiq committed:
Update/add llms.txt files for all sections of the documentation

piiq committed:
Revert a check that was supposed to prevent creating a file with the same name as a folder

piiq reviewed (approved):
I've added some changes. Namely:
- Changed OpenBB Platform to ODP and Open Data Platform in a few places in text, headers and keyword tags. I've left OpenBB Platform in the keywords for legacy SEO performance
- Updated the llms.txt files following the newer content that is generated from the current state of the docs and the main repo extensions
- Tested the broken URLs (all are fine)

There is one unresolved warning related to `fomc_documents` that I couldn't tackle:
When building I get this message:
```
[WARNING] Duplicate routes found!
- Attempting to create page at /python/reference/economy/fomc_documents, but a page already exists at this route.
This could lead to non-deterministic routing behavior.
```

This stems from the fact that the `generate_platform_markdown.py` creates both a `fomc_documents.md` and a `fomc_documents` folder. I was not able to immediately understand why it's happening even that I went and tried to use the route. I actually don't see the `economy.fomc_documents.download` command that is exposed as if `fomc_documents` is a folder. Will need @deeleeramone help here.

But given the state of the PR and the time constraints I don't see this as a blocking issue so I will merge this PR so that the docs are live.
